File: apps/backend/internal/dto/dict.go
package dto

// DictTypeQuery 字典类型列表查询条件。
type DictTypeQuery struct {
	PageQuery
	Name   string `form:"name" binding:"omitempty,max=64"`
	Type   string `form:"type" binding:"omitempty,max=64"`
	Status *int8  `form:"status" binding:"omitempty,oneof=0 1"`
}

// CreateDictTypeRequest 新增字典类型。
type CreateDictTypeRequest struct {
	Name string `json:"name" binding:"required,max=64"`
	// Type 是数据项的归属键，只允许字母/数字/下划线：
	// 它会作为 URL 路径段用于按类型取数据（/dicts/data/{type}）。
	Type   string `json:"type" binding:"required,max=64"`
	Status *int8  `json:"status" binding:"omitempty,oneof=0 1"`
	Remark string `json:"remark" binding:"omitempty,max=255"`
}

// UpdateDictTypeRequest 修改字典类型。
//
// Type 允许修改，但服务层会同步更新 sys_dict_data 的冗余列。
type UpdateDictTypeRequest struct {
	Name   *string `json:"name" binding:"omitempty,max=64"`
	Type   *string `json:"type" binding:"omitempty,max=64"`
	Status *int8   `json:"status" binding:"omitempty,oneof=0 1"`
	Remark *string `json:"remark" binding:"omitempty,max=255"`
}

// DictDataQuery 字典数据列表查询条件。
type DictDataQuery struct {
	PageQuery
	// DictType 为空时返回全部类型的数据；前端双栏布局下通常按类型过滤。
	DictType string `form:"dictType" binding:"omitempty,max=64"`
	Label    string `form:"label" binding:"omitempty,max=100"`
	Status   *int8  `form:"status" binding:"omitempty,oneof=0 1"`
}

// CreateDictDataRequest 新增字典数据项。
type CreateDictDataRequest struct {
	// DictTypeID 决定归属；dict_type 冗余列由服务层按此 ID 推导，不由前端传入，
	// 避免两者不一致。
	DictTypeID uint64 `json:"dictTypeId" binding:"required"`
	Label      string `json:"label" binding:"required,max=100"`
	Value      string `json:"value" binding:"required,max=100"`
	Sort       int    `json:"sort"`
	IsDefault  *int8  `json:"isDefault" binding:"omitempty,oneof=0 1"`
	Status     *int8  `json:"status" binding:"omitempty,oneof=0 1"`
	Remark     string `json:"remark" binding:"omitempty,max=255"`
}

// UpdateDictDataRequest 修改字典数据项。归属类型不可改。
type UpdateDictDataRequest struct {
	Label     *string `json:"label" binding:"omitempty,max=100"`
	Value     *string `json:"value" binding:"omitempty,max=100"`
	Sort      *int    `json:"sort"`
	IsDefault *int8   `json:"isDefault" binding:"omitempty,oneof=0 1"`
	Status    *int8   `json:"status" binding:"omitempty,oneof=0 1"`
	Remark    *string `json:"remark" binding:"omitempty,max=255"`
}

File: apps/web/src/api/dict.ts
/** 字典管理接口 */
import type { PageResult } from '@workbackend/shared'
import { http } from '@/utils/request'

/** 字典类型 */
export interface DictTypeItem {
  id: number
  name: string
  type: string
  status: number
  remark: string
  createdAt: string
}

/** 字典数据项 */
export interface DictDataItem {
  id: number
  dictTypeId: number
  dictType: string
  label: string
  value: string
  sort: number
  isDefault: number
  status: number
  remark: string
  createdAt: string
}

export interface DictTypeQuery {
  page?: number
  size?: number
  name?: string
  type?: string
  status?: number
}

export interface DictTypePayload {
  name?: string
  type?: string
  status?: number
  remark?: string
}

export interface DictDataQuery {
  page?: number
  size?: number
  dictType?: string
  label?: string
  status?: number
}

export interface DictDataPayload {
  /** 新增时必填；归属类型不可修改，故编辑时不传 */
  dictTypeId?: number
  label?: string
  value?: string
  sort?: number
  isDefault?: number
  status?: number
  remark?: string
}

// ---- 字典类型 ----

export function listDictTypes(params: DictTypeQuery) {
  return http.get<PageResult<DictTypeItem>>('/dicts/types', params)
}

export function createDictType(data: DictTypePayload) {
  return http.post<null>('/dicts/types', data)
}

export function updateDictType(id: number, data: DictTypePayload) {
  return http.put<null>(`/dicts/types/${id}`, data)
}

/** 删除类型会级联删除其下全部数据 */
export function deleteDictType(id: number) {
  return http.delete<null>(`/dicts/types/${id}`)
}

// ---- 字典数据 ----

export function listDictData(params: DictDataQuery) {
  return http.get<PageResult<DictDataItem>>('/dicts/data', params)
}

/**
 * 按类型取启用的字典项，供业务页面下拉框使用。
 * 路径含 /type 段是为了与 /data/:id 区分开（否则 gin 路由冲突）。
 */
export function getDictDataByType(dictType: string) {
  return http.get<DictDataItem[]>(`/dicts/data/type/${dictType}`)
}

export function createDictData(data: DictDataPayload) {
  return http.post<null>('/dicts/data', data)
}

export function updateDictData(id: number, data: DictDataPayload) {
  return http.put<null>(`/dicts/data/${id}`, data)
}

export function deleteDictData(id: number) {
  return http.delete<null>(`/dicts/data/${id}`)
}
